src/modules/dashboard/sagas/actions/meta/mintTokens.ts

```diff
@@ -0,0 +1,159 @@
+import { put, takeEvery } from 'redux-saga/effects';
+import { ClientType, ColonyClient, Network } from '@colony/colony-js';
+import { soliditySha3 } from 'web3-utils';
+import { splitSignature } from 'ethers/utils';
+import { hexSequenceNormalizer } from '@purser/core';
+
+import { getLoggedInUser } from '~data/index';
+import { ContextModule, TEMP_getContext } from '~context/index';
+import { Action, ActionTypes, AllActions } from '~redux/index';
+import { putError } from '~utils/saga/effects';
+import { DEFAULT_NETWORK } from '~constants';
+import { signMessage } from '../../../../core/sagas/messages';
+
+/*
+ * @TODO This is not a real action saga... yet
+ * It's just being used as an example to be able generate a basic metatransactions
+ *
+ * The rest of the content for this saga will be added by
+ * https://github.com/JoinColony/colonyDapp/pull/2779 which will also add a bunch
+ * of abstractions over some of this boilerplate code
+ */
+function* createMintTokensMetaAction({
+  payload: { colonyAddress, amount },
+  meta,
+}: Action<ActionTypes.META_MINT_TOKENS>) {
+  try {
+    if (!amount) {
+      throw new Error('Amount to mint not set for mintTokens transaction');
+    }
+    const { walletAddress: userAddress } = yield getLoggedInUser();
+
+    const colonyManager = TEMP_getContext(ContextModule.ColonyManager);
+    const { provider } = colonyManager;
+
+    const colonyClient: ColonyClient = yield colonyManager.getClient(
+      ClientType.ColonyClient,
+      colonyAddress,
+    );
+    const currentNonce = yield colonyClient.getMetatransactionNonce(
+      userAddress,
+    );
+
+    /*
+     * Prettier is the stupidest thing ever
+     */
+    // eslint-disable-next-line max-len
+    const encodedTransaction = colonyClient.interface.functions.mintTokens.encode(
+      [amount],
+    );
+
+    const { chainId: currentNetworkChainId } = yield provider.getNetwork();
```

Comment:
I could, but this file will be removed, as it's not really needed.

It was just put here so that we could test them via redux. 

The proper implementation doesn't require us to re-write all sagas with metatransactions in mind. The current existent sagas (without this one obviously) will just work both with normal transactions and with metatransactions.